src/test/java/com/google/cloud/spanner/pgadapter/ParserTest.java

```diff
@@ -685,6 +685,43 @@ public void testCopyParserOptions() throws Exception {
       assertThat(options.getFormat(), is(equalTo(Format.TEXT)));
       assertThat(options.getNullString(), is(equalTo("null")));
     }
+
+    {
+      char value;
+      for (value = '!'; value <= '~'; value++) {
+        CopyTreeParser.CopyOptions options = new CopyTreeParser.CopyOptions();
+        String sql = "COPY users FROM STDIN (QUOTE '" + value + "');";
+        parse(sql, options);
+        assertThat(options.getTableName(), is(equalTo("users")));
+        assertThat(options.getFromTo(), is(equalTo(FromTo.FROM)));
+        assertThat(options.getFormat(), is(equalTo(Format.TEXT)));
+        assertThat(options.getQuote(), is(equalTo(value)));
+      }
+    }
+
+    {
+      char value;
+      for (value = '!'; value <= '|'; value++) {
+        char value1 = (char) (value + 1);
+        char value2 = (char) (value + 2);
+        CopyTreeParser.CopyOptions options = new CopyTreeParser.CopyOptions();
+        String sql =
+            "COPY users FROM STDIN (DELIMITER '"
+                + value
+                + "', QUOTE '"
+                + value1
+                + "', ESCAPE '"
+                + value2
+                + "');";
+        parse(sql, options);
+        assertThat(options.getTableName(), is(equalTo("users")));
+        assertThat(options.getFromTo(), is(equalTo(FromTo.FROM)));
+        assertThat(options.getFormat(), is(equalTo(Format.TEXT)));
+        assertThat(options.getDelimiter(), is(equalTo(value)));
+        assertThat(options.getQuote(), is(equalTo(value1)));
+        assertThat(options.getEscape(), is(equalTo(value2)));
+      }
```

Comment:
Can we add the following assertion (here and also for the other tests):

```suggestion
        assertEquals(StatementParser.parseCommand(StatementParser.removeCommentsAndTrim(sql)), "COPY");
      }
```

This assertion currently fails for `ESCAPE '\'`, as our parser (correctly?) sees '\' as an unclosed string literal.

Reply:
Yes, removeCommentsAndTrim() does not allow a single '\\' in the statement.  Copy expects the delimiter, escape, and quote options to be single characters.

Le'ts drop ', ", and \ for now.  The logic in removeCommentsAndTrim() would break if we allowed those. We could add them back in at a later point after we rework removeCommentsAndTrim().

Reply:
yeah, that sounds reasonable.